src/main/scala/util/SynchronizerReg.scala

```diff
@@ -100,13 +107,17 @@ object AsyncResetSynchronizerShiftReg {
     apply (in, sync, init.litValue.toInt, None)
 }
 
+// Note: This module may end up with a non-Bool type reset.
+// But the Primitives within will always have Bool reset type.
 @deprecated("SyncResetSynchronizerShiftReg is unecessary with Chisel3 inferred resets. Use ResetSynchronizerShiftReg which will use the inferred reset type.", "rocket-chip 1.2")
```

Comment:
```suggestion
// Note: This module may end up with a non-Bool type reset.
// But the Primitives within will always have Bool reset type.
@deprecated("SyncResetSynchronizerShiftReg usage is discouraged. Use ResetSynchronizerShiftReg which will use the inferred reset type. Cast  the reset to Bool if you truly want to enforce SyncResetSynchronizerShiftReg functionality.", "rocket-chip 1.2")
```

Hmm, good point. What do you think of this instead (at least I should learn how to spell unnecessary)?